litestar/contrib/msgspec.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Generic, TypeVar, cast
+
+from msgspec import NODEFAULT, Struct, inspect
+
+from litestar.dto.factory.abc import AbstractDTOFactory
+from litestar.dto.factory.field import DTO_FIELD_META_KEY, DTOField
+from litestar.dto.factory.types import FieldDefinition
+from litestar.dto.factory.utils import get_model_type_hints
+from litestar.types.empty import Empty
+from litestar.utils.helpers import get_fully_qualified_class_name
+
+if TYPE_CHECKING:
+    from typing import Any, ClassVar, Collection, Generator
+
+    from litestar.typing import ParsedType
+
+__all__ = ("MsgspecDTO",)
+
+DataT = TypeVar("DataT", bound="Struct | Collection[Struct]")
+
+
+class MsgspecDTO(AbstractDTOFactory[DataT], Generic[DataT]):
+    """Support for domain modelling with Msgspec."""
+
+    __slots__ = ()
+
+    model_type: ClassVar[type[Struct]]
```

Comment:
Type is not allowed inside ClassVar, or is it now?! Big change if this passes mypy

Reply:
I think so!

```py
from __future__ import annotations

from typing import ClassVar


class A: ...


class Test:
    a: ClassVar[type[A]]


test = Test()
test.a = A()
```

```
main.py:14: error: Cannot assign to class variable "a" via instance  [misc]
main.py:14: error: Incompatible types in assignment (expression has type "A", variable has type "Type[A]")  [assignment]
```